src/test/compile-fail/auto-is-contextual.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+type A0 = auto;
+//~^ ERROR cannot find type `auto` in this scope
+type A1 = auto::auto;
+//~^ ERROR Use of undeclared type or module `auto`
+type A2 = auto<auto, auto>;
+//~^ ERROR cannot find type `auto` in this scope
+//~| ERROR cannot find type `auto` in this scope
+//~| ERROR cannot find type `auto` in this scope
+type A3 = auto<<auto as auto>::auto>;
+//~^ ERROR cannot find type `auto` in this scope
+//~| ERROR cannot find type `auto` in this scope
+//~| ERROR Use of undeclared type or module `auto`
+type A4 = auto(auto, auto) -> auto;
+//~^ ERROR cannot find type `auto` in this scope
+//~| ERROR cannot find type `auto` in this scope
+//~| ERROR cannot find type `auto` in this scope
+//~| ERROR cannot find type `auto` in this scope
```

Comment:
:)
I see this is copied from the `dyn Trait` PR.

All the examples here are in the "we are starting to parse a type" context, which is actual for `dyn Trait`, but not for `auto trait`.
See https://github.com/rust-lang/rust/blob/master/src/test/run-pass/union/union-backcomp.rs for better examples of what needs to be tested for a contextual keyword starting an item.

Reply:
if you're gonna steal, steal from the best! And also steal the right test.